docs/design/downward_api_resources_limits_requests.md

```diff
@@ -0,0 +1,537 @@
+   // Selects a key of a secret in the pod's namespace.
+   SecretKeyRef *SecretKeySelector `json:"secretKeyRef,omitempty"`
+}
+```
+
+#### Environment variables
+
+This table shows how selectors can be used for various requests and
+limits to be exposed as environment variables.
+
+| Name | Resource |
+| -------------------- | -------------------|
+| CPU_LIMIT | cpu_limit |
+| MEMORY_LIMIT | memory_limit |
+| CPU_REQUEST | cpu_request |
+|MEMORY_REQUEST | memory_request |
+
+Since environment variables are container scoped, it is optional
+to specify container name as part of the partial selectors as they are
+relative to container spec. If container name is not specified, then
+it defaults to current container. However, container name could be specified
+to expose variables from other containers.
+
+#### Volume plugin
+
+| Path | Resource |
+| -------------------- | -------------------|
+| cpu_limit | cpu_limit |
+| memory_limit | memory_limit|
+| cpu_request | cpu_request |
+| memory_request | memory_request |
+
+Since environment variables are container scoped, the container name must
+be specified.
+
+Note: Also, environment variables and volume path names are examples
+only and not necessarily as specified above.
+
+#### Examples
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: dapi-test-pod
+spec:
+  containers:
+    - name: test-container
+      image: gcr.io/google_containers/busybox
+      command: [ "/bin/sh","-c", "env" ]
+      resources:
+        requests:
+          memory: "64Mi"
+          cpu: "250m"
+        limits:
+          memory: "128Mi"
+          cpu: "500m"
+      env:
+        - name: CPU_LIMIT
+          valueFrom:
+            resourceFieldRef:
+              resource: cpu_limit
+```
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: kubernetes-downwardapi-volume-example
+spec:
+  containers:
+    - name: client-container
+      image: gcr.io/google_containers/busybox
+      command: ["sh", "-c","while true; do if [[ -e /etc/labels ]]; then cat /etc/labels; fi; if [[ -e /etc/annotations ]]; then cat /etc/annotations; fi; sleep 5; done"]
+      resources:
+        requests:
+          memory: "64Mi"
+          cpu: "250m"
+        limits:
+          memory: "128Mi"
+          cpu: "500m"
+      volumeMounts:
+        - name: podinfo
+          mountPath: /etc
+          readOnly: false
+  volumes:
+    - name: podinfo
+      downwardAPI:
+        items:
+          - path: "cpu_limit"
+            resourceFieldRef:
+              containerName: client-container
+              resource: cpu_limit
+```
+
+#### Validations
+
+For APIs with no selectors, verify that the resource strings are valid and is one
+of `cpu_limit`, `request_limit`, `cpu_request` and `memory_request`.
+Also verify that container name is provided with volumes.
+
+## Pod and container resource access
+Pod resources will always be accessed with `type ObjectFieldSelector` object in
```

Comment:
Like metdata.name, status.IP or any existing (not part of downward API yet) or new pod level resources in the future.